### ISSUE TITLE
AssertionError: 'c4_v220_span_corruption' does not exist in any of the task cache directories.

### ISSUE THREAD

**Issue by boram915** [google-research/t5x#923]
Hi,
I'm trying to verify the installation of T5 using the example code.
MODEL_DIR="${HOME}/test-model/"
python3 -m t5x.train \
  --gin_file=t5x/examples/t5/t5_1_1/examples/small_c4_pretrain.gin \
  --gin.MODEL_DIR=\"${MODEL_DIR}\" \
  --alsologtostderr

Then I get this error: AssertionError: 'c4_v220_span_corruption' does not exist in any of the task cache directories.

How I installed:
On the GCP TPU-VM v3-8 machine with  --version=tpu-vm-tf-2.10.0,
I step by step installed Jax, flax, t5[gcp] using pip (code given in the GCP userguide/tutorials)
and t5x (installation code given in the t5x offical github).
Installed T5 version is 0.9.3 
I checked t5.data.tasks.TaskRegistry().names() to see if c4_v220_span_corruption exists.

Full error message is below:
I1117 08:43:45.493216 140400864681024 resource_reader.py:50] system_path_file_exists:t5x/examples/t5/t5_1_1/small.gin
I1117 08:43:45.494145 140400864681024 resource_reader.py:37] gin-config opened resource file:/home/boram/t5x/t5x/examples/t5/t5_1_1/small.gin
I1117 08:43:45.494385 140400864681024 resource_reader.py:50] system_path_file_exists:t5x/examples/t5/t5_1_1/base.gin
I1117 08:43:45.494586 140400864681024 resource_reader.py:37] gin-config opened resource file:/home/boram/t5x/t5x/examples/t5/t5_1_1/base.gin
I1117 08:43:45.508894 140400864681024 resource_reader.py:50] system_path_file_exists:t5x/configs/runs/pretrain.gin
I1117 08:43:45.509320 140400864681024 resource_reader.py:37] gin-config opened resource file:/home/boram/t5x/t5x/configs/runs/pretrain.gin
I1117 08:43:45.526628 140400864681024 gin_utils.py:63] Gin Configuration:
I1117 08:43:45.530813 140400864681024 gin_utils.py:65] from __gin__ import dynamic_registration
I1117 08:43:45.530899 140400864681024 gin_utils.py:65] import __main__ as train_script
I1117 08:43:45.530941 140400864681024 gin_utils.py:65] import seqio
I1117 08:43:45.530974 140400864681024 gin_utils.py:65] import t5.data.mixtures
I1117 08:43:45.531011 140400864681024 gin_utils.py:65] import t5.data.tasks as t52
I1117 08:43:45.531057 140400864681024 gin_utils.py:65] from t5x import adafactor
I1117 08:43:45.531089 140400864681024 gin_utils.py:65] from t5x.examples.t5 import network
I1117 08:43:45.531118 140400864681024 gin_utils.py:65] from t5x import gin_utils
I1117 08:43:45.531145 140400864681024 gin_utils.py:65] from t5x import models
I1117 08:43:45.531173 140400864681024 gin_utils.py:65] from t5x import partitioning
I1117 08:43:45.531212 140400864681024 gin_utils.py:65] from t5x import trainer
I1117 08:43:45.531244 140400864681024 gin_utils.py:65] from t5x import utils
I1117 08:43:45.531301 140400864681024 gin_utils.py:65]
I1117 08:43:45.531354 140400864681024 gin_utils.py:65] # Macros:
I1117 08:43:45.531391 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.531452 140400864681024 gin_utils.py:65] BATCH_SIZE = 256
I1117 08:43:45.531489 140400864681024 gin_utils.py:65] DROPOUT_RATE = 0.0
I1117 08:43:45.531526 140400864681024 gin_utils.py:65] LABEL_SMOOTHING = 0.0
I1117 08:43:45.531564 140400864681024 gin_utils.py:65] LOSS_NORMALIZING_FACTOR = None
I1117 08:43:45.531608 140400864681024 gin_utils.py:65] MIXTURE_OR_TASK_MODULE = None
I1117 08:43:45.531645 140400864681024 gin_utils.py:65] MIXTURE_OR_TASK_NAME = 'c4_v220_span_corruption'
I1117 08:43:45.531682 140400864681024 gin_utils.py:65] MODEL = @models.EncoderDecoderModel()
I1117 08:43:45.531720 140400864681024 gin_utils.py:65] MODEL_DIR = '/home/boram/test-model/'
I1117 08:43:45.531757 140400864681024 gin_utils.py:65] OPTIMIZER = @adafactor.Adafactor()
I1117 08:43:45.531793 140400864681024 gin_utils.py:65] RANDOM_SEED = None
I1117 08:43:45.531830 140400864681024 gin_utils.py:65] SHUFFLE_TRAIN_EXAMPLES = True
I1117 08:43:45.531867 140400864681024 gin_utils.py:65] TASK_FEATURE_LENGTHS = {'inputs': 512, 'targets': 114}
I1117 08:43:45.531905 140400864681024 gin_utils.py:65] TRAIN_STEPS = 10000
I1117 08:43:45.531938 140400864681024 gin_utils.py:65] USE_CACHED_TASKS = True
I1117 08:43:45.531974 140400864681024 gin_utils.py:65] USE_HARDWARE_RNG = False
I1117 08:43:45.532011 140400864681024 gin_utils.py:65] VOCABULARY = @seqio.SentencePieceVocabulary()
I1117 08:43:45.532047 140400864681024 gin_utils.py:65] Z_LOSS = 0.0001
I1117 08:43:45.532086 140400864681024 gin_utils.py:65]
I1117 08:43:45.532123 140400864681024 gin_utils.py:65] # Parameters for adafactor.Adafactor:
I1117 08:43:45.532160 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.532197 140400864681024 gin_utils.py:65] adafactor.Adafactor.decay_rate = 0.8
I1117 08:43:45.532234 140400864681024 gin_utils.py:65] adafactor.Adafactor.logical_factor_rules = \
I1117 08:43:45.532274 140400864681024 gin_utils.py:65]     @adafactor.standard_logical_factor_rules()
I1117 08:43:45.532312 140400864681024 gin_utils.py:65] adafactor.Adafactor.step_offset = 0
I1117 08:43:45.532349 140400864681024 gin_utils.py:65]
I1117 08:43:45.532386 140400864681024 gin_utils.py:65] # Parameters for utils.CheckpointConfig:
I1117 08:43:45.532423 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.532460 140400864681024 gin_utils.py:65] utils.CheckpointConfig.restore = @utils.RestoreCheckpointConfig()
I1117 08:43:45.532497 140400864681024 gin_utils.py:65] utils.CheckpointConfig.save = @utils.SaveCheckpointConfig()
I1117 08:43:45.532536 140400864681024 gin_utils.py:65]
I1117 08:43:45.532573 140400864681024 gin_utils.py:65] # Parameters for utils.create_learning_rate_scheduler:
I1117 08:43:45.532615 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.532653 140400864681024 gin_utils.py:65] utils.create_learning_rate_scheduler.base_learning_rate = 1.0
I1117 08:43:45.532690 140400864681024 gin_utils.py:65] utils.create_learning_rate_scheduler.factors = 'constant * rsqrt_decay'
I1117 08:43:45.532729 140400864681024 gin_utils.py:65] utils.create_learning_rate_scheduler.warmup_steps = 10000
I1117 08:43:45.532766 140400864681024 gin_utils.py:65]
I1117 08:43:45.532803 140400864681024 gin_utils.py:65] # Parameters for train/utils.DatasetConfig:
I1117 08:43:45.532840 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.532877 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.batch_size = %BATCH_SIZE
I1117 08:43:45.532914 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.mixture_or_task_name = %MIXTURE_OR_TASK_NAME
I1117 08:43:45.532951 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.module = %MIXTURE_OR_TASK_MODULE
I1117 08:43:45.532988 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.pack = True
I1117 08:43:45.533025 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.seed = None
I1117 08:43:45.533063 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.shuffle = %SHUFFLE_TRAIN_EXAMPLES
I1117 08:43:45.533257 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.split = 'train'
I1117 08:43:45.533359 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.task_feature_lengths = %TASK_FEATURE_LENGTHS
I1117 08:43:45.533412 140400864681024 gin_utils.py:65] train/utils.DatasetConfig.use_cached = %USE_CACHED_TASKS
I1117 08:43:45.533471 140400864681024 gin_utils.py:65]
I1117 08:43:45.533510 140400864681024 gin_utils.py:65] # Parameters for train_eval/utils.DatasetConfig:
I1117 08:43:45.533548 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.533595 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.batch_size = %BATCH_SIZE
I1117 08:43:45.533632 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.mixture_or_task_name = %MIXTURE_OR_TASK_NAME
I1117 08:43:45.533669 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.module = %MIXTURE_OR_TASK_MODULE
I1117 08:43:45.533705 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.pack = True
I1117 08:43:45.533741 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.seed = 42
I1117 08:43:45.533777 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.shuffle = False
I1117 08:43:45.533816 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.split = 'validation'
I1117 08:43:45.533853 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.task_feature_lengths = %TASK_FEATURE_LENGTHS
I1117 08:43:45.533889 140400864681024 gin_utils.py:65] train_eval/utils.DatasetConfig.use_cached = %USE_CACHED_TASKS
I1117 08:43:45.533926 140400864681024 gin_utils.py:65]
I1117 08:43:45.533965 140400864681024 gin_utils.py:65] # Parameters for models.EncoderDecoderModel:
I1117 08:43:45.534002 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.534038 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.input_vocabulary = %VOCABULARY
I1117 08:43:45.534077 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.label_smoothing = %LABEL_SMOOTHING
I1117 08:43:45.534114 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.loss_normalizing_factor = %LOSS_NORMALIZING_FACTOR
I1117 08:43:45.534150 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.module = @network.Transformer()
I1117 08:43:45.534187 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.optimizer_def = %OPTIMIZER
I1117 08:43:45.534228 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.output_vocabulary = %VOCABULARY
I1117 08:43:45.534264 140400864681024 gin_utils.py:65] models.EncoderDecoderModel.z_loss = %Z_LOSS
I1117 08:43:45.534300 140400864681024 gin_utils.py:65]
I1117 08:43:45.534336 140400864681024 gin_utils.py:65] # Parameters for partitioning.PjitPartitioner:
I1117 08:43:45.534372 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.534408 140400864681024 gin_utils.py:65] partitioning.PjitPartitioner.logical_axis_rules = \
I1117 08:43:45.534444 140400864681024 gin_utils.py:65]     @partitioning.standard_logical_axis_rules()
I1117 08:43:45.534480 140400864681024 gin_utils.py:65] partitioning.PjitPartitioner.model_parallel_submesh = None
I1117 08:43:45.534516 140400864681024 gin_utils.py:65] partitioning.PjitPartitioner.num_partitions = 1
I1117 08:43:45.534552 140400864681024 gin_utils.py:65]
I1117 08:43:45.534686 140400864681024 gin_utils.py:65] # Parameters for utils.RestoreCheckpointConfig:
I1117 08:43:45.534731 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.534765 140400864681024 gin_utils.py:65] utils.RestoreCheckpointConfig.path = []
I1117 08:43:45.534798 140400864681024 gin_utils.py:65]
I1117 08:43:45.534830 140400864681024 gin_utils.py:65] # Parameters for utils.SaveCheckpointConfig:
I1117 08:43:45.534868 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.534901 140400864681024 gin_utils.py:65] utils.SaveCheckpointConfig.dtype = 'float32'
I1117 08:43:45.534933 140400864681024 gin_utils.py:65] utils.SaveCheckpointConfig.keep = None
I1117 08:43:45.534965 140400864681024 gin_utils.py:65] utils.SaveCheckpointConfig.period = 1000
I1117 08:43:45.534997 140400864681024 gin_utils.py:65] utils.SaveCheckpointConfig.save_dataset = False
I1117 08:43:45.535029 140400864681024 gin_utils.py:65]
I1117 08:43:45.535078 140400864681024 gin_utils.py:65] # Parameters for seqio.SentencePieceVocabulary:
I1117 08:43:45.535110 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.535142 140400864681024 gin_utils.py:65] seqio.SentencePieceVocabulary.sentencepiece_model_file = \
I1117 08:43:45.535174 140400864681024 gin_utils.py:65]     'gs://t5-data/vocabs/cc_all.32000.100extra/sentencepiece.model'
I1117 08:43:45.535207 140400864681024 gin_utils.py:65]
I1117 08:43:45.535239 140400864681024 gin_utils.py:65] # Parameters for network.T5Config:
I1117 08:43:45.535271 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.535303 140400864681024 gin_utils.py:65] network.T5Config.dropout_rate = %DROPOUT_RATE
I1117 08:43:45.535334 140400864681024 gin_utils.py:65] network.T5Config.dtype = 'bfloat16'
I1117 08:43:45.535367 140400864681024 gin_utils.py:65] network.T5Config.emb_dim = 512
I1117 08:43:45.535398 140400864681024 gin_utils.py:65] network.T5Config.head_dim = 64
I1117 08:43:45.535429 140400864681024 gin_utils.py:65] network.T5Config.logits_via_embedding = False
I1117 08:43:45.535461 140400864681024 gin_utils.py:65] network.T5Config.mlp_activations = ('gelu', 'linear')
I1117 08:43:45.535492 140400864681024 gin_utils.py:65] network.T5Config.mlp_dim = 1024
I1117 08:43:45.535524 140400864681024 gin_utils.py:65] network.T5Config.num_decoder_layers = 8
I1117 08:43:45.535556 140400864681024 gin_utils.py:65] network.T5Config.num_encoder_layers = 8
I1117 08:43:45.535595 140400864681024 gin_utils.py:65] network.T5Config.num_heads = 6
I1117 08:43:45.535627 140400864681024 gin_utils.py:65] network.T5Config.vocab_size = 32128
I1117 08:43:45.535658 140400864681024 gin_utils.py:65]
I1117 08:43:45.535690 140400864681024 gin_utils.py:65] # Parameters for train_script.train:
I1117 08:43:45.535722 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.535754 140400864681024 gin_utils.py:65] train_script.train.checkpoint_cfg = @utils.CheckpointConfig()
I1117 08:43:45.535786 140400864681024 gin_utils.py:65] train_script.train.eval_period = 1000
I1117 08:43:45.535817 140400864681024 gin_utils.py:65] train_script.train.eval_steps = 20
I1117 08:43:45.535848 140400864681024 gin_utils.py:65] train_script.train.infer_eval_dataset_cfg = None
I1117 08:43:45.535880 140400864681024 gin_utils.py:65] train_script.train.model = %MODEL
I1117 08:43:45.535912 140400864681024 gin_utils.py:65] train_script.train.model_dir = %MODEL_DIR
I1117 08:43:45.535943 140400864681024 gin_utils.py:65] train_script.train.partitioner = @partitioning.PjitPartitioner()
I1117 08:43:45.535975 140400864681024 gin_utils.py:65] train_script.train.random_seed = %RANDOM_SEED
I1117 08:43:45.536007 140400864681024 gin_utils.py:65] train_script.train.summarize_config_fn = @gin_utils.summarize_gin_config
I1117 08:43:45.536039 140400864681024 gin_utils.py:65] train_script.train.total_steps = %TRAIN_STEPS
I1117 08:43:45.536070 140400864681024 gin_utils.py:65] train_script.train.train_dataset_cfg = @train/utils.DatasetConfig()
I1117 08:43:45.536101 140400864681024 gin_utils.py:65] train_script.train.train_eval_dataset_cfg = @train_eval/utils.DatasetConfig()
I1117 08:43:45.536133 140400864681024 gin_utils.py:65] train_script.train.trainer_cls = @trainer.Trainer
I1117 08:43:45.536183 140400864681024 gin_utils.py:65] train_script.train.use_hardware_rng = %USE_HARDWARE_RNG
I1117 08:43:45.536223 140400864681024 gin_utils.py:65]
I1117 08:43:45.536267 140400864681024 gin_utils.py:65] # Parameters for trainer.Trainer:
I1117 08:43:45.536298 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.536329 140400864681024 gin_utils.py:65] trainer.Trainer.learning_rate_fn = @utils.create_learning_rate_scheduler()
I1117 08:43:45.536361 140400864681024 gin_utils.py:65] trainer.Trainer.num_microbatches = None
I1117 08:43:45.536392 140400864681024 gin_utils.py:65]
I1117 08:43:45.536423 140400864681024 gin_utils.py:65] # Parameters for network.Transformer:
I1117 08:43:45.536455 140400864681024 gin_utils.py:65] # ==============================================================================
I1117 08:43:45.536486 140400864681024 gin_utils.py:65] network.Transformer.config = @network.T5Config()
I1117 08:43:45.537348 140400864681024 partitioning.py:498] `activation_partitioning_dims` = 1, `parameter_partitioning_dims` = 1
I1117 08:43:49.975174 140400864681024 train.py:194] Process ID: 0
I1117 08:43:49.975799 140400864681024 train.py:198] GlobalDeviceArray enabled.
I1117 08:43:49.975952 140400864681024 train.py:233] Using fast RngBitGenerator PRNG for initialization and dropout.
I1117 08:43:50.064762 140400864681024 train.py:238] Random seed not provided, using RNG seed 1668674629
I1117 08:43:50.204496 140400864681024 partitioning.py:196] last device coords : (1, 1, 0, 1)
last local device coords: (1, 1, 0, 1)
W1117 08:43:50.204721 140400864681024 partitioning.py:242] Tiling device assignment mesh by hosts, which may lead to reduced XLA collective performance. To avoid this, modify the model parallel submesh or run with more tasks per host.
I1117 08:43:50.204898 140400864681024 partitioning.py:300] global_mesh axis_names: ('data', 'model')
I1117 08:43:50.204969 140400864681024 partitioning.py:301] global_mesh devices: [[TpuDevice(id=0, process_index=0, coords=(0,0,0), core_on_chip=0)]
 [TpuDevice(id=1, process_index=0, coords=(0,0,0), core_on_chip=1)]
 [TpuDevice(id=4, process_index=0, coords=(0,1,0), core_on_chip=0)]
 [TpuDevice(id=5, process_index=0, coords=(0,1,0), core_on_chip=1)]
 [TpuDevice(id=2, process_index=0, coords=(1,0,0), core_on_chip=0)]
 [TpuDevice(id=3, process_index=0, coords=(1,0,0), core_on_chip=1)]
 [TpuDevice(id=6, process_index=0, coords=(1,1,0), core_on_chip=0)]
 [TpuDevice(id=7, process_index=0, coords=(1,1,0), core_on_chip=1)]]
I1117 08:43:50.205403 140400864681024 partitioning.py:302] global_mesh devices shape: (8, 1)
I1117 08:43:52.970514 140400864681024 utils.py:1447] Initializing dataset for task 'c4_v220_span_corruption' with a replica batch size of 256 and a seed of 1668674632
I1117 08:43:52.970899 140400864681024 dataset_providers.py:1099] 'c4_v220_span_corruption' does not exist in any task cache directories (searched []).
I1117 08:43:52.970978 140400864681024 dataset_providers.py:1104] Using cache directory None for 'c4_v220_span_corruption'.
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/boram/t5x/t5x/train.py", line 753, in <module>
    gin_utils.run(main)
  File "/home/boram/t5x/t5x/gin_utils.py", line 107, in run
    app.run(
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/home/boram/t5x/t5x/train.py", line 713, in main
    _main(argv)
  File "/home/boram/t5x/t5x/train.py", line 749, in _main
    train_using_gin()
  File "/home/boram/.local/lib/python3.8/site-packages/gin/config.py", line 1605, in gin_wrapper
    utils.augment_exception_message_and_reraise(e, err_str)
  File "/home/boram/.local/lib/python3.8/site-packages/gin/utils.py", line 41, in augment_exception_message_and_reraise
    raise proxy.with_traceback(exception.__traceback__) from None
  File "/home/boram/.local/lib/python3.8/site-packages/gin/config.py", line 1582, in gin_wrapper
    return fn(*new_args, **new_kwargs)
  File "/home/boram/t5x/t5x/train.py", line 275, in train
    train_iter = get_dataset_fn(train_dataset_cfg, ds_shard_id, num_ds_shards,
  File "/home/boram/t5x/t5x/utils.py", line 1429, in get_dataset
    return get_dataset_inner(cfg, shard_info, feature_converter_cls, seed,
  File "/home/boram/t5x/t5x/utils.py", line 1451, in get_dataset_inner
    return seqio.get_dataset(
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1753, in get_dataset
    ds = mixture_or_task.get_dataset(
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1190, in get_dataset
    source = self._get_cached_source(split, file_shuffle_buffer_size)
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1254, in _get_cached_source
    self.assert_cached()
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1121, in assert_cached
    assert self.cache_dir, (
AssertionError: 'c4_v220_span_corruption' does not exist in any of the task cache directories.
  In call to configurable 'train' (<function train at 0x7fb0459393a0>)

Any comments would be greatly appreciated.

**Comment by fadebek1** [comment]
Try setting `USE_CACHED_TASKS = False`

**Comment by boram915** [comment]
> Try setting `USE_CACHED_TASKS = False`

Thank you. It worked. 
Though it caused out of memory error since I am using v3-8 without any additional disk attached. 
Somewhere I read it costs quite much to process the whole C4/en datasets, so I got processed one from huggingface and will try it.

I want to verify the installation t5x. So, I tried wmt_t2t_translate example. It caused another error belw:
$ python3 ${T5X_DIR}/t5x/train.py   --gin_file="t5x/examples/t5/t5_1_1/examples/base_wmt_from_scratch.gin"   --gin.MODEL_DIR=\"${MODEL_DIR}\"
...
I1122 08:40:15.243808 139680718634048 gin_utils.py:65] USE_CACHED_TASKS = False
I1122 08:40:20.680646 139680718634048 utils.py:1454] Initializing dataset for task 'wmt_t2t_ende_v003' with a replica batch size of 128 and a seed of 0
I1122 08:40:20.682885 139680718634048 dataset_providers.py:1671] Available task: wmt_t2t_ende_v003
Traceback (most recent call last):
  File "/home/boram/install/t5x/t5x/train.py", line 753, in <module>
    gin_utils.run(main)
  File "/home/boram/install/t5x/t5x/gin_utils.py", line 107, in run
    app.run(
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/home/boram/install/t5x/t5x/train.py", line 713, in main
    _main(argv)
  File "/home/boram/install/t5x/t5x/train.py", line 749, in _main
    train_using_gin()
  File "/home/boram/.local/lib/python3.8/site-packages/gin/config.py", line 1605, in gin_wrapper
    utils.augment_exception_message_and_reraise(e, err_str)
  File "/home/boram/.local/lib/python3.8/site-packages/gin/utils.py", line 41, in augment_exception_message_and_reraise
    raise proxy.with_traceback(exception.__traceback__) from None
  File "/home/boram/.local/lib/python3.8/site-packages/gin/config.py", line 1582, in gin_wrapper
    return fn(*new_args, **new_kwargs)
  File "/home/boram/install/t5x/t5x/train.py", line 275, in train
    train_iter = get_dataset_fn(train_dataset_cfg, ds_shard_id, num_ds_shards,
  File "/home/boram/install/t5x/t5x/utils.py", line 1432, in get_dataset
    return get_dataset_inner(cfg, shard_info, feature_converter_cls, seed,
  File "/home/boram/install/t5x/t5x/utils.py", line 1458, in get_dataset_inner
    return seqio.get_dataset(
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1738, in get_dataset
    mixture_or_task = get_mixture_or_task(mixture_or_task_name)
  File "/home/boram/.local/lib/python3.8/site-packages/seqio/dataset_providers.py", line 1674, in get_mixture_or_task
    raise ValueError(
ValueError: No Task or Mixture found with name '<seqio.dataset_providers.Task object at 0x7f0876fe01f0>'.
  In call to configurable 'train' (<function train at 0x7f087935e430>)
--------------------------------------------------------
Can I get some hints?

**Comment by boram915** [comment]
Becasue the error "ValueError: No Task or Mixture found with name '<seqio.dataset_providers.Task object" occurred with C4 and WMT,
I changed "t5x/utils.py" line 1465 
  from "mixture_or_task_name=mixture_or_task,"
  to "mixture_or_task_name=cfg.mixture_or_task_name,"
then seqio.dataset_providers.py seems to get the task as expected. Am I right and was that code wrong?

Now I struggles with USE_CACHED_TASKS option.
When I set it as "False", the code tried to download and prepare the C4 dataset from the tfds hub. Because it takes too long, I downloaded the prepared version of C4 from huggingface/allenai and uploaded it into GS bucket and used --tfds_data_dir='gs://my_bucket_name/datadir' with USE_CACHED_DIR=False.
Then "AssertionError: 'c4_v220_span_corruption' does not exist in any of the task cache directories." appeared again.
What would be the best solution in this case?
Why the t5x code doesn't properly work as it is documented with the example gin file?
Now I suspect the t5x code line by line. Is it natural to do so?
